### PR TITLE
Entryexitcontrollers

### PR DESCRIPTION
**Pull request by duane-mcmahon**
includes the google style xml in the commit, probably should have had put it in the gitignore. lolz

### CODE REVIEW

**Comment by WBGroupR** [comment]
Duane,

I need to delete my ‘carpark’ branch from the origin (remote) repository - but I don’t think i have the access to do it, I think only you do now. The reason why is that I was trying to do a push of my changes up to the remote yesterday so that I can create a pull request, but I thought that I had to first create the branch in the remote. But it turns out that the push is supposed to do that anyway. So now I have a useless empty branch called ‘carpark’ in the remote that needs to be deleted so I can do the push again.

So could you please delete that carpark branch in GitHub for me?

Thanks,
Wayne

**Comment by WBGroupR** [comment]
Hi Duane,

Forget the below. I’ve managed to get my carpark branch up to the remote, but had to do a lot to get it there. Still having issues with the .idea files for some reason.

Anyway, will now create a pull request.

Wayne

File: src/bcccp/carpark/entry/EntryController.java [hunk 1]
```diff
@@ -36,30 +34,34 @@ public EntryController(Carpark carpark, IGate entryGate, ICarSensor os, ICarSens
...
 
-  //adhoc
   @Override
   public void buttonPushed() {
 
-    adhocTicket = carpark.issueAdhocTicket();
+    IAdhocTicket adhocTicket = carpark.issueAdhocTicket();
 
-    carpark.recordAdhocTicketEntry();
+    ui.printTicket(carpark.getName(), adhocTicket.getTicketNo(), adhocTicket.getEntryDateTime(),
```
Comment (WBGroupR) [inline src/bcccp/carpark/entry/EntryController.java]: continuation of 3rd argument needs to be indented

File: src/bcccp/carpark/exit/ExitController.java [hunk 2]
```diff
@@ -20,40 +19,86 @@
...
-		
-	}
 
+		exitTime = new Date().getTime();
+
+		adhocTicket = carpark.getAdhocTicket(ticketStr);
+
+		if (adhocTicket != null) {
+
+			if (exitTime < (adhocTicket.getPaidDateTime() + FIFTEEN_MINUTES)) {
```
Comment (WBGroupR) [inline src/bcccp/carpark/exit/ExitController.java]: Block indentation should be +2 spaces, this appears to be 8 spaces.

File: src/bcccp/carpark/exit/ExitController.java [hunk 3]
```diff
@@ -20,40 +19,86 @@
...
+
+		outsideSensor = os;
 
+		this.ui = ui;
+
+	}
 
 
+	// STEP: Read barcode.
+// The bar code is read and a check is made that no more than 15 minutes have elapsed.
```
Comment (WBGroupR) [inline src/bcccp/carpark/exit/ExitController.java]: This comment line needs to be aligned with previous comment

**Review by WBGroupR — COMMENTED** [review]
EntryController:
2. Source File Basics - no issues found, input statements fine, ordering/spacing is fine
3. Source File Structure - no issues found - class contents ordered logically
4. Formatting - no issues found - braces are used where optional, K&R style non-empty blocks, block indentation +2 spaces, vertical and horizontal white space ok, declarations ok
5. Naming - no issues found, camelCase used
6. Programming Practices - no issues found
7. Javadoc - not used

ExitController:
2. Source File Basics - no issues found
3. Source File Structure - no issues found
4. Formatting - issues found as follows:
    Block indentation should be +2, but appears to be 8,      comment on line 42 should be intended to match same  block level. No other issues.
5. Naming - no issues found
6. Programming Practices - no issues found
7. Javadoc - none used

**Comment by WBGroupR** [comment]
Please see comments above - WB

File: src/bcccp/carpark/exit/ExitController.java (hunk 3 above)
Comment (duane-mcmahon) [reply src/bcccp/carpark/exit/ExitController.java]: i will format the code using the google style, ok. Chris can then action the merge.

**Comment by duane-mcmahon** [comment]
chris i have made another commit to make the code stylish, can you please merge this pull request. regards. I am assuming the merged branch will reflect the new commit I made in response to the review.

File: src/bcccp/carpark/exit/ExitController.java (hunk 3 above)
Comment (navajo00) [reply src/bcccp/carpark/exit/ExitController.java]: Please let me know when the code is fixed so Wayne can review it again. After the second review and if everything is OK, we can Merge with Master. 
Cheers,

**Review by navajo00 — APPROVED** [review]
Code style complies with Google Style guide.

**Comment by navajo00** [comment]
Wayne, there are conflicts in your branch. Could you please check the code and resolve them. 
Regards,
Christopher

**Pull request merged**